### PR TITLE
add friendlist

### PR DESCRIPTION
**Pull request by alex96dev**
- create api routes
- create User Scheme
- update homepage

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -160,17 +189,24 @@ export default function HomePage({ onSubmit }) {
...
-                category={activity.category}
-              />
-            </Link>
-          ))
+          displayedActivities
+            .filter(
+              (activity) =>
+                appUserFriendsList &&
+                appUserFriendsList.length > 0 &&
+                appUserFriendsList.find((user) => user === activity.authorId)
```
Comment (onemanwenttomow) [inline pages/index.js:197]: We could simplify this a bit with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). 

```suggestion
                appUserFriendsList?.find((user) => user === activity.authorId)
```

File: pages/index.js [hunk 2]
```diff
@@ -47,12 +50,38 @@ export default function HomePage({ onSubmit }) {
...
+
+  if (!activities) return <div>failed to load</div>;
+
+  if (activitiesIsLoading || status === "loading" || appUsersIsLoading)
+    return <div>loading...</div>;
+  if (!session) {
+    return (
+      <>
+        <LoginPage />
+      </>
```
Comment (onemanwenttomow) [inline pages/index.js:82]: We don't need the fragment if we only return one thing. 

```suggestion
        <LoginPage />
```

File: pages/index.js [hunk 3]
```diff
@@ -18,7 +18,10 @@ import useStore from "@/store";
 
 export default function HomePage({ onSubmit }) {
   const { data: session, status } = useSession();
-  const { data: activities, isLoading } = useSWR("/api/activities");
+  const { data: activities, isLoading: activitiesIsLoading } =
```
Comment (onemanwenttomow) [inline pages/index.js:21]: I think the fact that you have renamed the variable from `isLoading` to `activitiesIsLoading` will have knock on effects further down in your code. 

i.e. The code below references `isLoading` but that variable no longer exists. 

``` js
if (isLoading) return <div>loading...</div>;
```

In general it feels like you are handling the loading states multiple times. It is probably time to review this and tidy things up a bit.

File: pages/index.js [hunk 4]
```diff
@@ -47,12 +50,38 @@ export default function HomePage({ onSubmit }) {
...
         setCondition(condition);
+        if (appUsers) {
+          const user = appUsers.find((user) => user._id === session?.user?.id);
+
+          if (user && user.friends) {
+            setAppUserFriendsList(user.friends);
+          } else {
+            return;
+          }
+        }
```
Comment (onemanwenttomow) [inline pages/index.js:61]: We could also tidy this up a little

``` js
if (appUsers && session?.user?.id) {
  const user = appUsers.find(user => user._id === session.user.id);
  setAppUserFriendsList(user?.friends || []);
}
```

i would also advise setting the default state to be an empty array at first

``` js
const [appUserFriendsList, setAppUserFriendsList] = useState([]);
```

File: pages/friendlist.js [hunk 5]
```diff
@@ -4,9 +4,14 @@ import Logo from "@/Icons/Logo";
 import { theme } from "@/styles";
 import Navigation from "@/components/Navigation";
 import CardForm from "@/components/CardForm";
+import useAuthentication from "./api/auth/useAuthentication";
 import UserProfile from "@/components/UserProfile";
 
 export default function FriendList({ onSubmit, setIsEditMode }) {
+  const { authenticated, loading, session } = useAuthentication();
+  const userId = session?.user?.id;
+  const id = "65fd9d760c057e6bcdb880cd";
```
Comment (onemanwenttomow) [inline pages/friendlist.js:13]: Whilst i understand the need for test ids while building the project, i am not sure we want this merged into main. No need to make any changes this time.

**Review by onemanwenttomow — CHANGES_REQUESTED** [review]
Nice job team!

I added a few comments and suggestions. I would also take a look and try and work out if you really need the useEffect. Or if you can make the request for weather data using useSwr?

Maybe you need to separate the logic for the weather and the user friends. 

You also need to look into cleaning up the multiple loading states now that variable names have changed. Some of the old checks are now redundant.

**Comment by alex96dev** [comment]
Hey Pete!
I just implemented your suggestions. 

The idea of the test id in the friendlist.js is to already have a general logic for adding a friend id to the array of the current logged in user. We will continue working at the logic in a further story called "Fetch real data" in order to have a dynamic id.

**Pull request merged**